LateNight :: optional Big spinnies, various improvements

ronso0 opened the pull request:
- [x] choose Spinny, CoverArt or nothing (darker background makes Spinny indicator easier to recognize)
- [x] choose big or small Spinny/CoverArt, from ywwg/latenight-biggerspinny
- [x] add FX3/FX4 buttons to master mixer & Mic/Aux units
- [x] clean up tool bar
- [x] simplify skin setting buttons templates
- [x] increase minimum window size to 1366x768px (1302,668px to consider Ubuntu's vertical taskbar)
- [x] fix beatsize spinboxes for qt5
- [x] persistent visibility connections in lower half (Mic/Aux, Samplers) [Bug 1779575](https://bugs.launchpad.net/mixxx/+bug/1779575)
- [x] add Broadcast button to tool bar [Bug 1776258](https://bugs.launchpad.net/mixxx/+bug/1776258)
- [x] add LateNight launch image
- [x] add Load/Save Sampler Bank buttons
- [x] fix control vinyl status light
- [x] fix sampler VU meter graphics (wrong size of vector templates was exported to correct size in skin)
- [x] show CUE point in samplers' overview [Bug 1781830](https://bugs.launchpad.net/bugs/1781830)

![latenight-update](https://user-images.githubusercontent.com/5934199/43004973-aa18e1f4-8c31-11e8-9c14-cba48e1d2bf4.png)
(just an illustration..of course big/small spinny is toggled simultaneously on all decks)

New LateNight launch image:
![latenight-launch2](https://user-images.githubusercontent.com/5934199/43005064-fdfba48c-8c31-11e8-9373-c0e76c791849.png)

Please test and let's discuss if we increase the minimum size to 1366x768 or if we stick with 1280px screens.

ywwg committed:
LateNite: Reposition spinnies so the can be bigger

ywwg committed:
Make more room for the top bar

ronso0 committed:
LateNight: use simple COs to toggle all Spinny/CoverArt combinations, resize vector images

ronso0 committed:
LateNight: fix transport row & beatsize spinboxes

ronso0 committed:
LateNight: add FX3/4 to Master mixer, fix position of conditional 2-decks xFader buttons

ronso0 committed:
LateNight: adapt Skin menu to new COs, simplify skin menu button templates

ronso0 committed:
LateNight: add FX3/4 buttons to Mic/Aux

ronso0 committed:
LateNight: fix toolbar buttons, remove obsolete dividers

ronso0 committed:
LateNight: increase skin minimum size to 1344x668px

ronso0 committed:
LateNight: make BIG spinny/cover optional

nopeppermint commented:
@ronso0 looks good, thanks for keeping small cover Art ;-)

ronso0 committed:
LateNight: add Broadcast button to tool bar

ronso0 committed:
LateNight: simplify lower half (samplers, Mic/Aux, library), remove SizeAwareStacks & WidgetStacks

ronso0 commented:
> looks good, thanks for keeping small cover Art ;-)

yeah, both big and small use cases are valid IMO.

ronso0 committed:
LateNight: fix typos

ronso0 committed:
LateNight: add individual launch image

ywwg commented:
looks great thanks for picking this up!

ronso0 committed:
LateNight: grant Skin menu a little more room

ronso0 committed:
LateNight: style LaunchImage like VU meters

ronso0 committed:
LateNight: fix vinyl control status light

ronso0 committed:
LateNight: fix Sampler VU meter graphics

ronso0 commented:
Done, please test.

nopeppermint commented:
looks good

nopeppermint commented:
maybe move "load/save sampler bank" to submenu "Sampler" ?

ronso0 commented:
We don't have a Samplers menu. Maybe we'll have one when we add more samplers or other sampler options, but for now I don't want to open another submenu just for this two buttons.

ronso0 commented:
Is anyone using controls vinyl? I don't so I can't test related controls and the status light

ronso0 committed:
LateNight: show CUE point in samplers

nopeppermint commented:
>  that all buttons are dim when pressed.

which buttons are dimmed?

ronso0 commented:
all transport controls, rate buttons etc.
They are red/white text when active (i.e. HotCue is set) and get darker when you actually press them. IMO that doesn't match an action, i.e Play from cue, and I don't know of any hardware equivalent.
![latenight-pressed](https://user-images.githubusercontent.com/5934199/43035626-ab185446-8cf3-11e8-8824-e0448fffe5d6.png)

That's why I'd make the white text on 'active/set' buttons a tiny bit lesss bright, and the text/bg of such a pressed button brighter so pressed buttons stand up more.
![latenight-pressed](https://user-images.githubusercontent.com/5934199/43035774-80085ba0-8cf5-11e8-8648-a1c040668a1b.png)

ywwg commented:
> @ywwg It feels strange that all buttons are dim when pressed. I'd like to change the icons so that the text is a bit brighter than unpressed. Do you mind?

that's fine with me! I'd also be ok with replacing the pixmaps with font-based buttons, the design is a little out of place (as is the red color)

ywwg commented:
> Please test and let's discuss if we increase the minimum size to 1366x768 or if we stick with 1280px screens.

I think in the previous PR there was general consensus that 1366x768 would be ok.  My old 1280 laptop is long dead, which is what I designed the original to fit.  Did anyone else have an older machine that needed the smaller size?  (In general, I think we should save Shade for the smallest screens)

ronso0 committed:
Merge branch 'master' into latenight-BIG-spinny-skinmenu-toolbar

ronso0 commented:
rebased

ronso0 commented:
This is ready to merge.
The rest (fx mixmode buttons, buttons rework) will happen in another PR

ronso0 committed:
LateNight: flexible width for Preview label

ronso0 commented:
> preview deck text is cut off on my machine

don't know why this happens, font is included and shouldn't vary in size.
I replaced fixed size with `min,min` policy. Can you test again?

ywwg commented:
looks good now.  I do have the accessibility feature "large fonts" enabled, at 125%, and this sometimes causes issues like these.

ronso0 commented:
great! so this can be merged.

ywwg commented:
I think so, @daschuer any comments?

ronso0 committed:
LateNight: unify waveform markers

ronso0 committed:
LateNight: repair PreviewDeck vu meter graphics

ronso0 committed:
LateNight: remove hover effect from star rating in decks

nopeppermint commented:
@ronso0  why do you remove hover from star rating in decks?

ronso0 commented:
> @ronso0 why do you remove hover from star rating in decks?

After quite a while I used LateNight for an actual prep session. Check new tracks, sort them into crates or just delete them, rate them! This is were I found the hover effect disturbing, (=unexpected) in LateNight. Can't tell why I implemented it in the first place.. it doesn't match the skin anyway and it's not significant enogh to help, it's just disturbing IMO.
The actual star widget has a hover effect itself, so I removed it.

daschuer commented:
The minimum width has been increased. Now it does not fit next to the Ubuntu Unity task bar on my 1366x768 screen. This is not a strong requirement, but if we can make it fit again, it would be nice.

daschuer commented:
I found myself repeated clicking on the top right gear button for skin settings. 
Can we move the settings button there? We may also, or alternatively replace the gear button for the beatgrid with something else, a pencil icon, for "Edit beatgride" or something.

ronso0 commented:
> The minimum width has been increased. Now it does not fit next to the Ubuntu Unity task bar on my 1366x768 screen

Good point! So we should update the [skin_guidelines](https://www.mixxx.org/wiki/doku.php/skin_guidelines#the_optimal_size_for_skins) and add Ubuntu to the table.

> Can we move the settings button there?

Nope, the space above decks is reserved for collapsible waveforms only.
I can replace the gear with a pencil or a wrench, like I suggested for Maintenance in lib-redesign branch. Also I'll try to add the gear icon to the Settings button. It makes sense to have this consistent across skins.

daschuer commented:
Yes, thanks. My vote would be a pencil, because a wrench might be scared double left handed people. :-p

ronso0 committed:
LateNight: use wrench for beatgrid editor

ronso0 committed:
LateNight: add gear icon to Settings toggle

daschuer commented on res/skins/LateNight/skin.xml:
```diff
@@ -86,9 +88,35 @@
   <!-- MinimumSize should not be an exact monitor resolution. There needs
   to be space for the title bar or other chrome at full screen -->
-  <MinimumSize>1270,550</MinimumSize>
+  <MinimumSize>1344,668</MinimumSize>
```
1302,668 
for Full HD Displays running Ubunty Unity.

daschuer replied:
Than the EQ buttons are cutted with the Skin preferences visible though.
But the Mixxx is usable on my device now. 
If we consider the "preferences open" state as exceptional it should work.

ronso0 replied:
1366x768 is just HD (720+ lines)
actually I never really cared about squeezed skins with Pref pane open.
as long as the skin fits in its regular state where fine I think.

daschuer commented:
There is only the pending size issue, the rest LGTM

ronso0 commented:
> My vote would be a pencil, because a wrench might be scared double left handed people. :-p

ha! I know what you mean..and was curious how to translate the german phrase. dict.leo.org says
"folks how are all thumbs"
so let's remove the gear as well, and the faders, all that stuff that MUST be something technical

ronso0 committed:
LateNight: decrease minimum window size to fit on 1366x768 screens with default Ubuntu decorations/taskbar

Be-ing commented:
The minimum width and height are too wide to fit my 3840 x 2160 screen now (using Qt5's automatic scaling) :/

Be-ing commented:
I tried cherry-picking d8b5714 to 2.1 but without the skin settings menu, there isn't enough horizontal space in the toolbar...

ronso0 commented:
> there isn't enough horizontal space in the toolbar...

Whaaat?
![latenight_toolbarspacer](https://user-images.githubusercontent.com/5934199/43781053-54f683c8-9a5c-11e8-9002-6440eb012532.png)

It's not realistic to test all skins with X screen resolutions and if automatic scaling fails so badly with 4K, we should bring back the possibility to opt out IMO

Be-ing commented:
There is enough horizontal space in the toolbar for the broadcast icon in 2.2, but 2.1 has more buttons in the toolbar because it does not have the skin settings menu.

Automatic scaling works well. Why was the minimum size increased in this PR? LateNight worked fine on my screen before. The minimum size should be as small as possible, not just to fit in the widest variety of screen sizes, but also so Mixxx can be run more comfortably in a window that is not full screen.

ronso0 commented:
LateNight 2.1.0 Besides the question why you would import the broadcast button into 2.1, I see there's enough space for it at minimal window size:
![latenight-2 1 0](https://user-images.githubusercontent.com/5934199/43809530-eb08dbd0-9ab2-11e8-9441-0298ec12b101.png)


> The minimum width and height are too wide to fit my 3840 x 2160 screen now (using Qt5's automatic scaling) :/

Please post a screenshot to show what is squeezed. Or is it just the warning in Prefences > Interface?
And yes, I consider Qt Scaling failing if it can't put a skin designed for 1366px on a 4K screen.

The minimum size was increased after we (incl. you) decided to do so in #1734.
This way we can have bigger spinnies AND a usable artist/title box.

> also so Mixxx can be run more comfortably in a window that is not full screen.

That's why I decreased it after @daschuer noticed it wouldn't fit next to the Ubuntu taskbar.

ronso0 commented:
Damn, I now notice this should actuallly go to 2.2 to have consitent Spinny/CoverArt controls.
@Be-ing @daschuer Do you agree? If yes, how to merge this into 2.2?

Also I'll soon open a small-fix PR to clean up the spinny COs in Deere so they really match those implemented here that should go to 2.2, as well.

daschuer commented:
> Edit just pulled 2.2 and noitce this is already included...?

Yes, currently 2.2 and master are the same.
